src/test/coins_tests.cpp

```diff
@@ -324,7 +324,7 @@ BOOST_AUTO_TEST_CASE(updatecoins_simulation_test)
             tx.vout.resize(1);
             tx.vout[0].nValue = i; //Keep txs unique unless intended to duplicate
             tx.vout[0].scriptPubKey.assign(InsecureRand32() & 0x3F, 0); // Random sizes so we can test memory usage accounting
-            unsigned int height = InsecureRand32();
+            const int height{int(InsecureRand32() >> 1)};
```

Comment:
use we just use `GetRandInt` from random.h?

Reply:
Might be better to do this in a separate pull request unrelated from the changes here?

Reply:
meh, what's the point of a new PR just for that? Probably no-one would review it 😂 imo, it makes much more sense here

Reply:
`GetRandInt` is either non-deterministic, or returns a constant (deterministic). Can you explain why either change makes sense? I'd say neither does and it should be left-as is.

Reply:
Ahh, okay, that's a good reason :) Maybe in the future we should adjust `InsecureRand32` to make it clearer that it can be deterministic